python/pyspark/sql/streaming/python_streaming_source_runner.py

```diff
@@ -210,7 +210,8 @@ def main(infile: IO, outfile: IO) -> None:
     # Read information about how to connect back to the JVM from the environment.
     java_port = int(os.environ["PYTHON_WORKER_FACTORY_PORT"])
     auth_secret = os.environ["PYTHON_WORKER_FACTORY_SECRET"]
-    (sock_file, _) = local_connect_and_auth(java_port, auth_secret)
+    (sock_file, sock) = local_connect_and_auth(java_port, auth_secret)
+    sock.settimeout(None)
```

Comment:
nit: can we add a short comment here on why we need to set this timeout?

Reply:
comment added.